sql/src/main/java/io/crate/analyze/MultiSourceSelect.java

```diff
@@ -37,49 +37,23 @@
 public class MultiSourceSelect implements QueriedRelation {
 
     private final RelationSplitter splitter;
-
-    public static class Source {
-        private final AnalyzedRelation relation;
-        private final QuerySpec querySpec;
-
-        public Source(AnalyzedRelation relation, QuerySpec querySpec) {
-            this.relation = relation;
-            this.querySpec = querySpec;
-        }
-
-        public QuerySpec querySpec() {
-            return querySpec;
-        }
-
-        public AnalyzedRelation relation() {
-            return relation;
-        }
-
-        @Override
-        public String toString() {
-            return "Source{" +
-                   "rel=" + relation +
-                   ", qs=" + querySpec +
-                   '}';
-        }
-    }
-
     private final HashMap<QualifiedName, Source> sources;
-    private final List<Field> fields;
     private final QuerySpec querySpec;
+    private final Fields fields;
 
     public MultiSourceSelect(
             Map<QualifiedName, AnalyzedRelation> sources,
-            List<? extends Path> outputNames,
+            Collection<? extends Path> paths,
```

Comment:
I think outputNames was a better name here.
Paths is just a repetition of the information already available. It doesn't tell anybody how these paths are used or what they are for